omniscidb/IR/Node.h

```diff
@@ -591,7 +591,11 @@ class TranslatedJoin : public Node {
     CHECK(false);
     return nullptr;
   }
-  const std::string& getFieldName(size_t i) const override { CHECK(false); }
+  const std::string& getFieldName(size_t i) const override {
+    CHECK(false);
+    // just a stub, shouldn't be reached.
+    return op_type_;
```

Comment:
Returning a reference to a local object would produce another warning though. I wonder if we can modify `CHECK` in such a way that the compiler would understand what we actually call `abort`. Or maybe use some new macro like `FAIL` to avoid unreachable return statements.